blueocean-github-pipeline/src/test/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubOrgFolderTest.java

```diff
@@ -1,48 +1,126 @@
 
-        put("/organizations/jenkins/pipelines/jenkinsci/",
-                ImmutableMap.of("name", "jenkinsci",
+        new RequestBuilder(baseUrl)
+                .status(200)
+                .jwtToken(getJwtToken(j.jenkins,"bob", "bob"))
+                .put("/organizations/jenkins/pipelines/jenkinsci/")
+                .data(ImmutableMap.of("name", "jenkinsci",
                         "$class", "io.jenkins.blueocean.blueocean_github_pipeline.GithubPipelineUpdateRequest",
                         "scmConfig", ImmutableMap.of("config",
                                 ImmutableMap.of("repos", ImmutableList.of("stapler")))
-                ), 200);
+                ))
+                .build(Map.class);
+    }
+
+    @Test
+    public void shouldFindUserStoreCredential() throws IOException {
+        //add username password credential to user's credential store in user domain and in USER scope
+        User user = login();
+        CredentialsStore store=null;
+        for(CredentialsStore s: CredentialsProvider.lookupStores(user)){
+            if(s.hasPermission(CredentialsProvider.CREATE) && s.hasPermission(CredentialsProvider.UPDATE)){
+                store = s;
+                break;
+            }
+        }
+        store.addDomain(new Domain("github-domain",
+                "Github Domain to store personal access token",
+                Collections.<DomainSpecification>emptyList()
+        ));
+        Domain domain = store.getDomainByName("github-domain");
+        StandardUsernamePasswordCredentials credential = new UsernamePasswordCredentialsImpl(CredentialsScope.USER,
+                "github", "Github Access Token", user.getId(), "12345");
+        store.addCredentials(domain, credential);
+
+
+        //create org folder and attach user and credential id to it
+        OrganizationFolder organizationFolder = j.createProject(OrganizationFolder.class, "demo");
+        AbstractFolderProperty prop = new BlueOceanCredentialsProvider.FolderPropertyImpl(user.getId(), credential.getId(), "github-domain",
+                ImmutableList.<DomainRequirement>of(new BlueOceanDomainRequirement()));
+        organizationFolder.addProperty(prop);
+
+        // lookup for created credential id in system store, it should resolve to previously created user store credential
+        StandardCredentials c = Connector.lookupScanCredentials(organizationFolder, null, credential.getId());
+        assertNotNull(c);
+        assertTrue(c instanceof StandardUsernamePasswordCredentials);
+        StandardUsernamePasswordCredentials usernamePasswordCredentials = (StandardUsernamePasswordCredentials) c;
+        assertEquals(credential.getId(), usernamePasswordCredentials.getId());
+        assertEquals(credential.getPassword().getPlainText(),usernamePasswordCredentials.getPassword().getPlainText());
+        assertEquals(credential.getUsername(),usernamePasswordCredentials.getUsername());
```

Comment:
Missing a test where there is a credential with the same ID in the system store, verifying that your proxy store is a higher ordinal and hence will get asked first